batch.go

```diff
@@ -124,6 +126,39 @@ func (t *Batch) Add(nd Node) error {
 	return t.err
 }
 
+// Remove removes a node from the DAGService. This is a non-batching
+// operation.
+func (t *Batch) Remove(ctx context.Context, c cid.Cid) error {
```

Comment:
This needs to remove the node from the queue (or flush the queue) and then remove the node.